overlord/devicestate/devicemgr.go

```diff
@@ -599,6 +615,52 @@ func (m *DeviceManager) ensureOperational() error {
 	return nil
 }
 
+// maybeRestoreAfterReset attempts to restore the serial assertion with a
+// matching key in a post-factory reset scenario. It is possible that it is
+// called when the device was unregistered, but when doing so, the device key is
+// removed.
+func (m *DeviceManager) maybeRestoreAfterReset(device *auth.DeviceState) (keyID, serial string, err error) {
```

Comment:
my preference is for this to return:  (*asserts.Serial, error),  then the caller can do .Serial() and .DeviceKey().ID() as needed

Reply:
updated